File: leetcode_ly/java/Array/MinimumSizeSubarraySum.java
public class Solution {
    public int minSubArrayLen(int s, int[] nums) {
        if(nums.length == 0) return 0;
        if(nums[0] >= s) return 1;
        int start = 0;
        int end = 1;
        int sum = nums[0];
        int len = Integer.MAX_VALUE;
        while(start <= end && end < nums.length){
            sum += nums[end];
            if(sum >= s){
                while(sum >= s){
                    sum -= nums[start];
                    start++;
                }
                len = Math.min(len,end - start + 2);
            }
            end++;
        }
        return (len == Integer.MAX_VALUE) ? 0 : len;
    }
}

File: leetcode_ly/c++/Array/MinimumSizeSubarraySum.cpp
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        int start = 0;
        int end = 0;
        int sum = 0;
        int minLen = 2147483647;
        
        while(start < nums.size() && end < nums.size()){
            while(sum < s && end < nums.size()){
                sum += nums[end];
                end++;
            }
            while(sum >= s && start <= end){
                sum -= nums[start];
                start++;
                int len = end - start + 1;
                minLen = (len < minLen) ? len : minLen;
            }
        }
        return (minLen == 2147483647) ? 0 : minLen;
    }
};